File: src/courses/data/catalog/02-dsa/dsa-core/implementations/csharp/sorting.cs
/**
 * C# Implementation: Sorting (Quick, Merge, Heap)
 * Invariant: Comparison O(N log N), Space O(1) to O(N)
 */
using System;
using System.Collections.Generic;

namespace Seed.DSA.Core
{
    public class SortingQuickMergeHeap<T>
    {
        private readonly List<T> _items = new List<T>();

        // TODO: Implement algorithm operations
        public void Add(T item)
        {
            _items.Add(item);
        }

        public int Count => _items.Count;
    }
}

File: src/courses/data/catalog/02-dsa/dsa-core/implementations/java/sorting.java
/**
 * Java Implementation: Sorting (Quick, Merge, Heap)
 * Invariant: Comparison O(N log N), Space O(1) to O(N)
 */
package seed.dsa.core;

import java.util.*;

public class SortingQuickMergeHeap<T> {
    private final List<T> items = new ArrayList<>();

    // TODO: Implement your solution here
    public void add(T value) {
        items.add(value);
    }

    public int size() {
        return items.size();
    }
}
